### PR TITLE
Fix modifier lookup

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/codegen/ir/IRGenerator.cpp [hunk 1]
```diff
@@ -338,14 +338,23 @@ string IRGenerator::generateModifier(
 		t("retParams", joinHumanReadable(retParams));
 		t("assignRetParams", assignRetParams);
 
-		solAssert(*_modifierInvocation.name().annotation().requiredLookup == VirtualLookup::Virtual, "");
-
-		ModifierDefinition const& modifier = dynamic_cast<ModifierDefinition const&>(
+		ModifierDefinition const* modifier = &dynamic_cast<ModifierDefinition const&>(
 			*_modifierInvocation.name().annotation().referencedDeclaration
```
Comment (mijovic) [inline libsolidity/codegen/ir/IRGenerator.cpp]: Maybe better to cast to pointer directly?

```suggestion
		auto const* modifier = dynamic_cast<ModifierDefinition const*>(
 			_modifierInvocation.name().annotation().referencedDeclaration
```

**Review by mijovic — COMMENTED** [review]
Looks good. Maybe just a small suggestion for a cast.

File: libsolidity/codegen/ir/IRGenerator.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/codegen/ir/IRGenerator.cpp]: If we do that, I have to add another solAssert, but I can do that.

**Comment by chriseth** [comment]
Updated.

**Review by mijovic — DISMISSED** [review]

**Pull request merged**